### PR TITLE
Fix file permission error on Appveyor, other clean-ups

### LINKED ISSUE #267: Example breaks windows testing
#263 has enable doctests and as a result the appveyor build is complaining file access error as it's being used by another process. (The weird thing is that all tests pass, so it would be nice to be able to turn this strictness off).

The workaround I ended up with is to allow the failures of the appveyor jobs to let the release procedure for 0.4 proceed.

cc @pllim @astrofrog @saimn as you are the ones who solved similar issues in astropy before.

```
============= 955 passed, 8 skipped, 13 xfailed in 35.62 seconds ==============
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "C:\conda\envs\test\lib\site-packages\astropy\utils\decorators.py", line 860, in test
    func = make_function_with_signature(func, name=name, **wrapped_args)
  File "C:\conda\envs\test\lib\site-packages\astropy\tests\runner.py", line 260, in test
    return runner.run_tests(**kwargs)
  File "C:\conda\envs\test\lib\site-packages\astropy\tests\runner.py", line 605, in run_tests
    return super().run_tests(**kwargs)
  File "C:\conda\envs\test\lib\site-packages\astropy\tests\runner.py", line 242, in run_tests
    return pytest.main(args=args, plugins=plugins)
  File "C:\conda\envs\test\lib\site-packages\astropy\config\paths.py", line 182, in __exit__
    shutil.rmtree(self._path)
  File "C:\conda\envs\test\lib\shutil.py", line 513, in rmtree
    return _rmtree_unsafe(path, onerror)
  File "C:\conda\envs\test\lib\shutil.py", line 392, in _rmtree_unsafe
    _rmtree_unsafe(fullname, onerror)
  File "C:\conda\envs\test\lib\shutil.py", line 392, in _rmtree_unsafe
    _rmtree_unsafe(fullname, onerror)
  File "C:\conda\envs\test\lib\shutil.py", line 392, in _rmtree_unsafe
    _rmtree_unsafe(fullname, onerror)
  File "C:\conda\envs\test\lib\shutil.py", line 397, in _rmtree_unsafe
    onerror(os.unlink, fullname, sys.exc_info())
  File "C:\conda\envs\test\lib\shutil.py", line 395, in _rmtree_unsafe
    os.unlink(fullname)
PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\\Users\\appveyor\\AppData\\Local\\Temp\\1\\tmppexf4t80astropy_cache\\astropy\\download\\py3\\2c9202ae878ecfcb60878ceb63837f5f'
```

### PR DESCRIPTION
**Pull request by pllim**
Fix #267

### CODE REVIEW

**Comment by pllim** [comment]
If still fails, need to hack `astropy/config/paths.py` to print out more info.

**Comment by pllim** [comment]
The pesky `C:\\Users\\appveyor\\AppData\\Local\\Temp\\1\\tmp73zkfjyxastropy_cache\\astropy\\download\\py3\\2c9202ae878ecfcb60878ceb63837f5f` ! It looks like I need to run this locally after all, so I can look at what is inside the offending hash. :weary:

**Comment by pllim** [comment]
**Notes from local Windows run investigation**

`pytest` was caching something to temporary directory. On Windows, this is usually `C:\Users\<username>\AppData\Local\Temp\<hash>astropy_cache\astropy\download\py3`. Note that this is different from the regular `astropy` cache location, which would be something like `C:\Users\<username>\.astropy\cache`.

We know the failure is from `<hash>astropy_cache` and not the regular `.astropy\cache` directory. We know the `<hash>` changes with every `pytest` run. So, I tested this package locally on Windows to get the failure (and failed it did). From the traceback, I obtained the `<hash>` for that run, then I see what were being cached with the following code:

```python
import shelve

# The offending cache dir. The trailing urlmap is important.
urlmapfn = 'C:\\...\\download\\py3\\urlmap'

# Inspect the cache.
with shelve.open(urlmapfn) as url2hash:
    print(list(url2hash.items()))
```

And with this, I saw two things inside:

1. http://data.astropy.org/tutorials/FITS-images/HorseHead.fits (used in `docs/plot_reg.py`) -- Hash for this file matched the exact hash in traceback
2. http://data.astropy.org/photometry/M6707HH.fits (used in `docs/masks.rst`)

So, now we know which example was crashing the doctest. But why? A search brought us to https://github.com/astropy/astropy/issues/7404#issuecomment-387874448 . And indeed, setting `memmap=False` to the code that reads `HorseHead.fits` finally fixed the problem on Windows.

File: docs/masks.rst [hunk 1]
```diff
@@ -160,6 +161,7 @@ as well as a cutout with the data multiplied by the mask::
     >>> mask = aperture.to_mask(mode='exact')
     >>> data = mask.cutout(hdu.data)
     >>> weighted_data = mask.multiply(hdu.data)
+    >>> pf.close()
```
Comment (pllim) [inline docs/masks.rst]: I closed some dangling file pointers. These don't fix any errors but I think it is good practice anyway.

File: docs/plot_reg.py [hunk 2]
```diff
@@ -7,7 +7,7 @@
 
 image_file = get_pkg_data_filename('tutorials/FITS-images/HorseHead.fits')
 print(image_file)
-image_data = fits.getdata(image_file, ext=0)
+image_data = fits.getdata(image_file, ext=0, memmap=False)
```
Comment (pllim) [inline docs/plot_reg.py]: This is the meat of the PR.

File: regions/_geometry/core.pxd [hunk 3]
```diff
@@ -1,4 +1,5 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
+# cython: language_level=3
```
Comment (pllim) [inline regions/_geometry/core.pxd]: Also fixed some Cython warnings I saw when installing the package from source. Follow up of #270

File: regions/io/fits/tests/test_fits_region.py [hunk 4]
```diff
@@ -1,9 +1,6 @@
-
-import distutils.version as vers
```
Comment (pllim) [inline regions/io/fits/tests/test_fits_region.py]: Removed unused imports.

File: regions/io/fits/write.py [hunk 5]
```diff
@@ -65,11 +65,13 @@ def write_fits_region(filename, regions, header=None):
 
     >>> from astropy.utils.data import get_pkg_data_filename
     >>> from astropy.io import fits
-    >>> file_sample = get_pkg_data_filename('data/fits_region.fits', package='regions.io.fits.tests')
+    >>> file_sample = get_pkg_data_filename(
+    ...     'data/fits_region.fits', package='regions.io.fits.tests')
```
Comment (pllim) [inline regions/io/fits/write.py]: Indentation makes doc prettier.

**Comment by cdeil** [comment]
@pllim - Thank you very much for your expertise and time to fix this!

**Comment by bsipocz** [comment]
@pllim - you're a  hero, thanks for this fix!

**Pull request merged**